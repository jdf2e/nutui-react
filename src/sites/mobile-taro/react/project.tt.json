{
  "miniprogramRoot": "./",
  "projectname": "first",
  "appid": "testAppId",
  "setting": {
    "es6": false,
    "minified": false
  }
}
