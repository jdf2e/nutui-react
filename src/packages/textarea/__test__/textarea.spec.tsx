// import * as renderer from 'react-test-renderer'
import * as React from 'react'
import '@testing-library/jest-dom'
import { render, fireEvent } from '@testing-library/react'
import { useState } from 'react'
import TextArea from '@/packages/textarea'

test('textarea props test', () => {
  const { container } = render(
    <TextArea defaultValue="基本用法" limitshow maxLength={20} />
  )
  expect(container.querySelector('.nut-textarea__textarea')?.innerHTML).toBe(
    '基本用法'
  )
  expect(container.querySelector('.nut-textarea__limit')?.innerHTML).toBe(
    '4/20'
  )
  expect(container).toMatchSnapshot()
})

test('textarea readOnly test', () => {
  const { container } = render(<TextArea readOnly />)
  expect(container.querySelector('.nut-textarea__textarea')).toHaveAttribute(
    'readonly'
  )
})

test('textarea disabled test', () => {
  const { container } = render(<TextArea disabled />)
  expect(container.querySelector('.nut-textarea__textarea')).toHaveAttribute(
    'disabled'
  )
  expect(container.querySelector('.nut-textarea')).toHaveClass(
    'nut-textarea__disabled'
  )
})

test('textarea onChange event test', () => {
  const handleChange = jest.fn()
  const TextareaDemo = () => {
    const [value, setValue] = useState('文字')
    return (
      <TextArea
        data-testid="textarea"
        defaultValue={value}
        onChange={handleChange}
      />
    )
  }

  const { container } = render(<TextareaDemo />)
  expect(container.querySelector('.nut-textarea__textarea')?.innerHTML).toBe(
    '文字'
  )
  const textareaEl = container.querySelector(
    '.nut-textarea__textarea'
  ) as Element
  fireEvent.change(textareaEl, { target: { value: '文字改变' } })
  expect(handleChange).toBeCalled()
  expect(container.querySelector('.nut-textarea__textarea')?.innerHTML).toBe(
    '文字改变'
  )
})

test('textarea onFocus event test', () => {
  const handleFocus = jest.fn()
  const handleBlur = jest.fn()
  const TextareaDemo = () => {
    return (
      <TextArea
        data-testid="textarea"
        onFocus={handleFocus}
        onBlur={handleBlur}
      />
    )
  }

  const { container } = render(<TextareaDemo />)

  const textareaEl = container.querySelector(
    '.nut-textarea__textarea'
  ) as Element
  fireEvent.focus(textareaEl)
  expect(handleFocus).toBeCalled()
  fireEvent.blur(textareaEl)
  expect(handleBlur).toBeCalled()
})
