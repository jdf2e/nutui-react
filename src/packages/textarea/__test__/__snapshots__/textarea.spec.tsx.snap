// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textarea props test 1`] = `
<div>
  <div
    class="nut-textarea"
  >
    <textarea
      class="nut-textarea__textarea"
      maxlength="20"
      placeholder="请输入内容"
      rows="2"
    >
      基础用法
    </textarea>
    <div
      class="nut-textarea__limit"
    >
      4
      /
      20
    </div>
  </div>
</div>
`;
